### DISCUSSION TITLE
testing custom client on hubs cloud

### DISCUSSION THREAD

**Discussion by blairmacintyre** [mozilla/hubs#4188] [Hubs Cloud Questions]
I'm following the (nice, simple) instructions to build and test a client, against my hubs-cloud instance.

I build and run with either `npm run dev` or `npm run start` after I login with `npm run login`

Going to `https://localhost:8080` shows me my sites front page.  The room likes are still our real URL (i.e., `https://hubs.aelatgt.net/...` but if I change the URL to go to localhost:8080, it drops be back to the main page.

I don't know why.   Very little appears in the console.  It's as if the hubs cloud server doesn't feed back the page at all.

**Comment by netpro2k** [comment]
Since you are only running a local copy of the client (and not reticulum) you can't visit the same urls as you do on the actual hubs cloud instance. To test a specific room you would should go to https://localhost:8080/hub.html?hub_id=XXXXXXX. Clicking the "Create Room" button on the homepage should land you on this page as well.

You will also need to make sure your hubs cloud instance allows connections from localhost:8080 in the admin console under CORS settings. **Take care when changing this setting to include your main domain as well or you can lock yourself out of the admin console.**

**Comment by blairmacintyre** [comment]
Ok, so this worked.  And I resolved three.js errors with `npm ci` (I didn't think I'd run npm install on this repo before, but who knows).  Now, I get problems with not being able to drop things on the windows:
![image](https://user-images.githubusercontent.com/1348861/116083239-666b8a00-a66a-11eb-8ccd-4dfcaa36ee30.png)

We have localhost:* in a bunch of the -src rules.  I'm pretty sure it's what's missing in the CORS settings.  But, the dire warning about getting locked out scares me ... :slight_smile:

If my domain is hubs.aelatgt.net and my link domain is aelatgt.link, what specifically should I put in the CORS settings?  

hubs.aelatgt.net, aelatgt.link, localhost:*

Or something else?
